test/torchaudio_unittest/models/tacotron2/model_test.py

```diff
@@ -0,0 +1,202 @@
+
+import torch
+from torchaudio.models.tacotron2 import _Tacotron2, _Encoder, _Decoder
+from torchaudio_unittest.common_utils import (
+    TorchaudioTestCase,
+    skipIfNoCuda,
+    TempDirMixin,
+)
+from parameterized import parameterized
+
+
+class TorchscriptConsistencyMixin(TempDirMixin):
+    """Mixin to provide easy access assert torchscript consistency"""
+
+    def _assert_torchscript_consistency(self, model, tensors):
+        path = self.get_temp_path('func.zip')
+        torch.jit.script(model).save(path)
+        ts_func = torch.jit.load(path)
+
+        torch.random.manual_seed(40)
+        output = model(*tensors)
+
+        torch.random.manual_seed(40)
+        ts_output = ts_func(*tensors)
+
+        self.assertEqual(ts_output, output)
+
+
+class TestEncoder(TorchaudioTestCase, TorchscriptConsistencyMixin):
+
+    def _get_model(self, encoder_n_convolutions=3, encoder_embedding_dim=512, encoder_kernel_size=5):
+        return _Encoder(encoder_n_convolutions, encoder_embedding_dim, encoder_kernel_size)
+
+    def _output_shape_test(self, model, n_batch, encoder_embedding_dim, n_seq, device, dtype):
+        """Validate the output dimensions of a Encoder.
+        """
+
+        x = torch.rand(n_batch, encoder_embedding_dim, n_seq, device=device, dtype=dtype)
+        input_lengths = torch.ones(n_batch, device=device, dtype=torch.int32) * n_seq
+        out = model(x, input_lengths)
+
+        assert out.size() == (n_batch, n_seq, encoder_embedding_dim)
+
+    @parameterized.expand([(32, 64, 512, torch.float32, torch.device('cpu'))])
+    def test_cpu_encoder_output(self, n_batch, n_seq, encoder_embedding_dim, dtype, device):
```

Comment:
Please write a short script of what is "intended" in this test.
For example; "A should meet condition B".

Running `pytest --collect-only` will give this one-liner as the summary of the tests.

Reply:
I've added more comments for each of the test functions (here)[https://github.com/pytorch/audio/pull/1621/files#diff-001793fb337f116c66a3e7f3869a18012a618ab079df93db381fec0f45b63d9eR29], but I am not sure why they didn't show up on my `pytest --coolect-only`

Reply:
offline chat with @mthrok. `pytest --collect-only -v` will do the job.